src/Service/userAction.js

```diff
@@ -0,0 +1,184 @@
+import axios from 'axios'
+
+export const isInHistory = (history , video) => {
+    return history.some(el => el._id === video._id)
+}
+
+export const isInLikedVideos = (likedVideos , video) => {
+    return likedVideos.some(el => el._id === video._id)
+}
+export const isInWatchLater = (watchlater , video) => {
+    return watchlater.some(el => el._id === video._id)
+}
+
+export const isInPlayList = (playlist , video) => {
+    return playlist.videos.some(el => el._id === video._id)
+}
+
+export const addToHistory = async (dispatchUserData , token , video) => {
+    try {
+       const {data , status} = await axios.post('/api/user/history',{
+            video
+        },{
+            headers : {
+                authorization : token
+            }
+        })
+        if(status === 200 || status === 201)
+          dispatchUserData({type : "ADD_TO_HISTORY" , payload : video})
+    }catch(error){
+        console.log(error)
+    }
+}
+export const removeFromHistory = async(dispatchUserData , token , video) => {
+    try{
+       const {data , status } = await axios.delete(`/api/user/history/${video._id}`,{
+            headers : {
+                authorization : token
+            }
+        })
+        if(status === 200){
+            dispatchUserData({type : "REMOVE_FROM_HISTORY" , payload : video})
```

Comment:
can use variables to avoid typo bugs

Reply:
I will at end make all reducer action type constants